### PR TITLE
migrate some more resource to new resource handling

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | no
| ⚠ Deprecations?   | yes
| ❌ BC Break       | yes
| 🔗 Related issues | #347
| ❓ Documentation  | no

## Description

migrate sqs-queue
migrate sqs_queue_policy and sqs_default_queue_policy
migrate subnet and default subnet
 migrate vpc

### CODE REVIEW

**Review by eliecharra — COMMENTED** [review]
Some small remarks

File: pkg/middlewares/aws_sqs_queue_policy_expander_test.go [hunk 1]
```diff
@@ -20,85 +20,212 @@ func TestAwsSqsQueuePolicyExpander_Execute(t *testing.T) {
...
+					Attrs: &resource.Attributes{
+						"queue_url": "foo",
+						"id":        "foo",
+						"policy":    "{\"Id\":\"MYINLINESQSPOLICY\",\"Statement\":[{\"Action\":\"sqs:SendMessage\",\"Effect\":\"Allow\",\"Principal\":\"*\",\"Resource\":\"arn:aws:sqs:eu-west-3:047081014315:foo\",\"Sid\":\"Stmt1611769527792\"}],\"Version\":\"2012-10-17\"}",
+					},
 				},
 			},
+			func(factory *terraform.MockResourceFactory) {
+				factory.On("CreateAbstractResource", "aws_sqs_queue_policy", "foo", mock.MatchedBy(func(input map[string]interface{}) bool {
+					return input["id"] == "foo"
```
[the remark this answers is not in the document]
Comment (moadibfr) [reply pkg/middlewares/aws_sqs_queue_policy_expander_test.go:57]: no there is only one resource in this test I think you are misreading resourcesFromState and expected?

[the remark this answers is not in the document]
Comment (eliecharra) [reply pkg/middlewares/aws_sqs_queue_policy_expander_test.go]: Yeah I inverted with expected my bad

File: pkg/middlewares/aws_sqs_queue_policy_expander_test.go [hunk 2]
```diff
@@ -20,85 +20,212 @@ func TestAwsSqsQueuePolicyExpander_Execute(t *testing.T) {
...
+					Id:   "foo",
+					Type: aws.AwsSqsQueuePolicyResourceType,
+					Attrs: &resource.Attributes{
+						"id":        "foo",
+						"queue_url": "foo",
+						"policy":    "{\"Id\":\"MYSQSPOLICY\",\"Statement\":[{\"Action\":\"sqs:SendMessage\",\"Effect\":\"Allow\",\"Principal\":\"*\",\"Resource\":\"arn:aws:sqs:eu-west-3:047081014315:foo\",\"Sid\":\"Stmt1611769527792\"}],\"Version\":\"2012-10-17\"}",
+					},
 				},
+			}, func(factory *terraform.MockResourceFactory) {
+				factory.On("CreateResource", mock.Anything, "aws_sqs_queue_policy").Once().Return(nil, nil)
```
[the remark this answers is not in the document]
Comment (moadibfr) [reply pkg/middlewares/aws_sqs_queue_policy_expander_test.go]: seems like I missed one...

File: pkg/resource/aws/aws_sqs_queue_policy.go [hunk 3]
```diff
@@ -23,3 +27,11 @@ func (r *AwsSqsQueuePolicy) TerraformType() string {
 func (r *AwsSqsQueuePolicy) CtyValue() *cty.Value {
 	return r.CtyVal
 }
+
+func initAwsSqsQueuePolicyMetaData(resourceSchemaRepository resource.SchemaRepositoryInterface) {
+	resourceSchemaRepository.UpdateSchema(AwsSqsQueuePolicyResourceType, map[string]func(attributeSchema *resource.AttributeSchema){
+		"policy": func(attributeSchema *resource.AttributeSchema) {
+			attributeSchema.JsonString = true
+		},
+	})
```
Comment (wbeuil) [inline pkg/resource/aws/aws_sqs_queue_policy.go:36]: Where is the normalization of the json string that we did on the `_ext` file ?

```go
resourceSchemaRepository.SetNormalizeFunc(AwsSqsQueuePolicyResourceType, func(val *resource.Attributes) {
  jsonString, err := helpers.NormalizeJsonString((*val)["policy"])
  if err != nil {
    return
  }
  val.SafeSet([]string{"policy"}, jsonString)
})
```

File: pkg/resource/aws/aws_vpc_test.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	"testing"
+
+	"github.com/cloudskiff/driftctl/test"
+
+	"github.com/cloudskiff/driftctl/test/acceptance"
+)
+
+func TestAcc_AwsVPC(t *testing.T) {
+	acceptance.Run(t, acceptance.AccTestCase{
+		Paths: []string{"./testdata/acc/aws_vpc"},
```
Comment (wbeuil) [inline pkg/resource/aws/aws_vpc_test.go:13]: It misses the `TerraformVersion: "X.X.X"` arg in the `AccTestCase`

File: pkg/resource/aws/metadatas.go [hunk 5]
```diff
@@ -26,4 +26,7 @@ func InitResourcesMetadata(resourceSchemaRepository resource.SchemaRepositoryInt
 	initAwsIAMUserMetaData(resourceSchemaRepository)
 	initAwsIAMUserPolicyMetaData(resourceSchemaRepository)
 	initAwsKeyPairMetaData(resourceSchemaRepository)
+	initAwsSubnetMetaData(resourceSchemaRepository)
+	initAwsDefaultSubnetMetaData(resourceSchemaRepository)
```
Comment (wbeuil) [inline pkg/resource/aws/metadatas.go:30]: This is a duplicate of the one line 11.

File: pkg/resource/resource.go [hunk 6]
```diff
@@ -67,10 +67,12 @@ var refactoredResources = []string{
...
 	"aws_sns_topic_subscription",
-	// "aws_sqs_queue",
-	// "aws_sqs_queue_policy",
-	// "aws_subnet",
-	// "aws_vpc",
+	"aws_sqs_queue",
+	"aws_sqs_queue_policy",
+	"aws_subnet",
+	"aws_subnet",
+	"aws_default_subnet",
```
Comment (wbeuil) [inline pkg/resource/resource.go:74]: Please remove one `aws_subnet` and there is already the `aws_default_subnet` line 28

**Review by wbeuil — CHANGES_REQUESTED** [review]

**Review by wbeuil — APPROVED** [review]

**Review by eliecharra — APPROVED** [review]

**Pull request merged**